File: src/fractal/world/fractum/_fractum.py
"""

    *Fractum*

  A fractum is a quantum and a spectrum.

"""

from abc import ABCMeta
from dataclasses import dataclass

from ._ident import FractumIdent


@dataclass
class Fractum(
    metaclass=ABCMeta,
):
    ident: tuple[int, ...]
    position: tuple[int, ...]
    quantum: Quantum
    spectrum: Spectrum

    def __hash__(self) -> int:
        return hash(self.ident)

File: cogn_fractum/src/fractum/fractum.rs
//! A fractum is a quantum extended with a spectrum.
//!
//! The name comes from the self-referential nature that this creates.

pub struct Fractum {
    position: Position,
    quantum: Quantum,
    spectrum: Spectrum,
}

impl Hash for Fractum {
    // fn __hash__(&self) -> int:
    //     return hash(&self.ident)
}
